### PR TITLE
chore : update overview sections to use real data(CXSPA-3136)

### PR DESCRIPTION
**Pull request by kimhw0630**
Closes: https://jira.tools.sap/browse/CXSPA-3136

- Overview header section 
- Active cart section
- Saved cart section
- Interests section
- common components clean up

### CODE REVIEW

File: feature-libs/asm/customer-360/components/customer-360/customer-360.component.html [hunk 1]
```diff
@@ -18,63 +18,112 @@ <h3 id="asm-customer-360-title" class="modal-title">
...
-          </div>
+          <ng-container
+            *ngIf="customerOverview?.userAvatar?.url; else customerTextAvatar"
+          >
+            <div
+              class="cx-avatar-image"
+              *ngIf="customerOverview?.userAvatar?.url"
+            >
+              <cx-media
+                [container]="getAvatarImage(customerOverview)"
```
Comment (rmch91) [inline feature-libs/asm/customer-360/components/customer-360/customer-360.component.html]: Usage of methods inside of template is not good from performance point of view. Method will execute all the time when, change detection will run. Better to use pure pipe, which will be triggered only when the source will be changed. Maybe you can create pipe just for this component (if this functionality is used only here) and move logic from method to the pipe?

File: feature-libs/asm/customer-360/components/customer-360/customer-360.component.html [hunk 2]
```diff
@@ -18,63 +18,112 @@ <h3 id="asm-customer-360-title" class="modal-title">
...
+              ></cx-media>
+            </div>
+          </ng-container>
+
+          <ng-template #customerTextAvatar>
+            <div class="cx-avatar" *ngIf="!customerOverview?.userAvatar?.url">
+              {{ getAvatarText() }}
```
Comment (rmch91) [inline feature-libs/asm/customer-360/components/customer-360/customer-360.component.html]: same as above

File: feature-libs/asm/customer-360/components/customer-360/customer-360.component.html [hunk 3]
```diff
@@ -18,63 +18,112 @@ <h3 id="asm-customer-360-title" class="modal-title">
...
+            </div>
+          </ng-template>
+
           <div class="cx-customer-info ml-3">
             <div class="cx-customer-name">
               {{
                 'customer360.header.subTitle'
-                  | cxTranslate: { name: customer.firstName }
+                  | cxTranslate: { name: customerOverview.name }
```
Comment (rmch91) [inline feature-libs/asm/customer-360/components/customer-360/customer-360.component.html]: maybe customerOverview?.name ?

File: feature-libs/asm/customer-360/components/customer-360/customer-360.component.html [hunk 4]
```diff
@@ -18,63 +18,112 @@ <h3 id="asm-customer-360-title" class="modal-title">
...
               }}
             </div>
             <div class="cx-customer-email">
-              {{ customer.uid }}
+              {{ customerOverview.email }}
```
Comment (rmch91) [inline feature-libs/asm/customer-360/components/customer-360/customer-360.component.html]: customerOverview?.email

File: feature-libs/asm/customer-360/components/customer-360/customer-360.component.html [hunk 5]
```diff
@@ -18,63 +18,112 @@ <h3 id="asm-customer-360-title" class="modal-title">
...
             </div>
           </div>
         </div>
         <div class="header-account-details">
           <span
             class="header-account-details-active-cart"
-            *ngIf="activeCart$ | async as activeCart"
+            *ngIf="customerOverview.cartCode"
```
Comment (rmch91) [inline feature-libs/asm/customer-360/components/customer-360/customer-360.component.html]: customerOverview?.cartCode

File: feature-libs/asm/customer-360/components/customer-360/customer-360.component.html [hunk 6]
```diff
@@ -18,63 +18,112 @@ <h3 id="asm-customer-360-title" class="modal-title">
...
           >
             <cx-icon class="account-icon" [type]="cartIcon"></cx-icon>
             {{
               'customer360.header.activeCartLabel'
-                | cxTranslate: { cartSize: activeCart.totalItems }
+                | cxTranslate: { cartSize: customerOverview.cartSize }
```
Comment (rmch91) [inline feature-libs/asm/customer-360/components/customer-360/customer-360.component.html]: There are places where you are using `customerOverview?.something` and somewhere you are not using question mark. So if we not sure that `customerOverview` always will exist, maybe we should add the question mark everywhere

File: feature-libs/asm/customer-360/root/model/customer-360.model.ts [hunk 7]
```diff
@@ -130,43 +131,48 @@ export interface Customer360CartEntry {
...
 
 export interface Customer360Avatar {
   url?: string;
   format?: string;
 }
 
-export interface Customer360Overview {
-  type: Customer360Type.OVERVIEW;
+export interface CustomerOverview {
+  address: Address;
```
Comment (rmch91) [inline feature-libs/asm/customer-360/root/model/customer-360.model.ts]: should it be optional also?

**Review by rmch91 — CHANGES_REQUESTED** [review]

File: feature-libs/asm/customer-360/components/customer-360/customer-360.component.html (hunk 1 above)
Comment (kimhw0630) [reply feature-libs/asm/customer-360/components/customer-360/customer-360.component.html]: thanks created a pipe.

File: feature-libs/asm/customer-360/components/customer-360/avatar-image.pipe.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Pipe, PipeTransform } from '@angular/core';
+import { CustomerOverview } from '@spartacus/asm/customer-360/root';
+import { Image } from '@spartacus/core';
+
+@Pipe({ name: 'avatarImage' })
```
Comment (rmch91) [inline feature-libs/asm/customer-360/components/customer-360/avatar-image.pipe.ts]: Nitpick, but almost all of pipes names in the storefront starts from "cx".  Maybe for consistency we should add "cx" prefix also.

File: feature-libs/asm/customer-360/components/customer-360/avatar-label.pipe.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Pipe, PipeTransform } from '@angular/core';
+import { User } from '@spartacus/core';
+
+@Pipe({ name: 'avatarLabel' })
```
Comment (rmch91) [inline feature-libs/asm/customer-360/components/customer-360/avatar-label.pipe.ts]: Nitpick, but almost all of pipes names in the storefront starts from "cx".  Maybe for consistency we should add "cx" prefix also.

**Review by rmch91 — APPROVED** [review]

**Pull request merged**